rules/aws_resource_missing_tags.go

```diff
@@ -77,29 +150,67 @@ func (r *AwsResourceMissingTagsRule) Check(runner tflint.Runner) error {
 		}
 
 		resources, err := runner.GetResourceContent(resourceType, &hclext.BodySchema{
-			Attributes: []hclext.AttributeSchema{{Name: tagsAttributeName}},
+			Attributes: []hclext.AttributeSchema{
+				{Name: tagsAttributeName},
+				{Name: providerAttributeName},
+			},
 		}, nil)
 		if err != nil {
 			return err
 		}
 
+		if resources.IsEmpty() {
+			continue
+		}
+
 		for _, resource := range resources.Blocks {
-			if attribute, ok := resource.Body.Attributes[tagsAttributeName]; ok {
-				logger.Debug("Walk `%s` attribute", resource.Labels[0]+"."+resource.Labels[1]+"."+tagsAttributeName)
-				wantType := cty.Map(cty.String)
-				err := runner.EvaluateExpr(attribute.Expr, func(resourceTags map[string]string) error {
+			providerName := "default"
+			if val, ok := resource.Body.Attributes[providerAttributeName]; ok {
+        if diagnostics := gohcl.DecodeExpression(val.Expr, nil, &providerName); diagnostics.HasErrors() {
+          logger.Error("Error decoding provider: %w", diagnostics)
+			    return diagnostics
+        }
+
+				if err != nil {
+					return err
+				}
+			}
+			resourceTags := make(map[string]string)
+
+			// The provider tags are to be overriden
+			// https://registry.terraform.io/providers/hashicorp/aws/latest/docs#default_tags
+			maps.Copy(resourceTags, providerTagsMap[providerName])
+
+			// If the resource has a tags attribute
+			if attribute, okResource := resource.Body.Attributes[tagsAttributeName]; okResource {
+				logger.Debug(
+					"Walk `%s` attribute",
+					resource.Labels[0]+"."+resource.Labels[1]+"."+tagsAttributeName,
+				)
+				// Since the evlauateExpr, overrides k/v pairs, we need to re-copy the tags
+				resourceTagsAux := make(map[string]string)
+
+        if diagnostics := gohcl.DecodeExpression(attribute.Expr, nil, &resourceTagsAux); diagnostics.HasErrors() {
```

Comment:
We should use `EvaluateExpr` here. The `DecodeExpression` without `EvalContext` will fail to evaluate variables like `var.foo` because it doesn't take variables into account. On the other hand, `EvaluateExpr` can evaluate this correctly.

Reply:
Fixed